java-checks/src/main/java/org/sonar/java/checks/ArrayDesignatorAfterTypeCheck.java

```diff
@@ -47,13 +49,25 @@ public void visitNode(Tree tree) {
       ArrayTypeTree arrayTypeTree = (ArrayTypeTree) returnType;
       SyntaxToken openBracketToken = arrayTypeTree.openBracketToken();
       if (isInvalidPosition(openBracketToken, identifierToken)) {
-        reportIssue(openBracketToken, "Move the array designators \"[]\" to the end of the return type.");
+        QuickFixHelper.newIssue(context)
+          .forRule(this)
+          .onTree(openBracketToken)
+          .withMessage("Move the array designators \"[]\" to the end of the return type.")
+          .withQuickFix(() -> createQuickFixes(arrayTypeTree))
+          .report();
         break;
       }
       returnType = arrayTypeTree.type();
     }
   }
 
+  private static JavaQuickFix createQuickFixes(ArrayTypeTree type) {
+    return JavaQuickFix.newQuickFix("Move [] to the method type")
+      .addTextEdit(JavaTextEdit.replaceBetweenTree(type.openBracketToken(), type.closeBracketToken(), ""))
```

Comment:
Maybe we could add a new method `JavaTextEdit.removeBetweenTree(Tree start, Tree end)` to avoid having to type `""` all the time.